flask/views.py

```diff
@@ -117,6 +117,9 @@ def __init__(cls, name, bases, d):
 
         if 'methods' not in d:
             methods = set()
+            for base in bases:
+                if base.__dict__.get('methods'):
+                    methods = methods.union(base.methods)
```

Comment:
can we simply use `methods.add()`, since you don't need `method` to do anything within the loop.

Reply:
Could you explain more? I don't quite get it. Thanks!

Reply:
```python
if hasattr(base, 'methods'):
    methods.update(base.methods)
```

Reply:
Yes, that does make more sense! Thanks, @jab!